evm/src/main/java/com/horizen/evm/interop/EvmLog.java

```diff
@@ -1,23 +1,22 @@
 package com.horizen.evm.interop;
 
-import com.fasterxml.jackson.annotation.JsonSetter;
-import com.fasterxml.jackson.annotation.Nulls;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.horizen.evm.utils.Address;
 import com.horizen.evm.utils.Converter;
 import com.horizen.evm.utils.Hash;
 
 import java.util.Arrays;
 
 public class EvmLog {
-    public Address address;
+    private Address address = Address.addressZero();
+    private Hash[] topics = new Hash[0];
+    private byte[] data = new byte[0];
 
-    @JsonSetter(nulls = Nulls.SKIP)
-    public Hash[] topics = new Hash[0];
-    @JsonSetter(nulls = Nulls.SKIP)
-    public byte[] data = new byte[0];
+    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+    public EvmLog(@JsonProperty("address") Address address, @JsonProperty("topics") Hash[] topics, @JsonProperty("data") byte[] data) {
 
-    public EvmLog(Address address, Hash[] topics, byte[] data) {
-        this.address = address;
+        this.address = address == null ? Address.addressZero() : address;
```

Comment:
Topics can be zero in case of anonymous events, data can be zero if only event only contains indexed args (topics).
If address is null, that means it is actually AddressZero (that is the way it is implemented on the GO side), this should probably be changed.